### PR TITLE
avm2: Stub flash.system.Capabilities

### PR DESCRIPTION
**Pull request by rdrpenguin04**
Depends on Ragnarok.

Adds stubs for more commonly-used functions in `flash.system.Capabilities`; specifically, the ones used in the SWF on the About Adobe Flash page.

### CODE REVIEW

File: core/src/avm2.rs [hunk 1]
```diff
@@ -102,7 +102,7 @@ impl<'gc> Avm2<'gc> {
             broadcast_list: HashMap::new(),
 
             #[cfg(feature = "avm_debug")]
-            debug_output: false,
+            debug_output: true,
```
Comment (adrian17) [inline core/src/avm2.rs:105]: don't forget to remove this :)

**Comment by danielhjacobs** [comment]
> https://github.com/ruffle-rs/ruffle/pull/4879/commits/3f405a7cbd4d5db23fa9314e914dd74819c47e77#diff-7822a3eb7478100e4687627162508ec020c0de25ad5113d6d8c2994e5f9b6a0bR54

I know it's temporary, but did you mean LNX? The prefix LIN is not listed: https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/system/Capabilities.html#version

**Comment by Herschel** [comment]
Could you rebase onto master? Thank you!

**Comment by rdrpenguin04** [comment]
> > [3f405a7#diff-7822a3eb7478100e4687627162508ec020c0de25ad5113d6d8c2994e5f9b6a0bR54](https://github.com/ruffle-rs/ruffle/commit/3f405a7cbd4d5db23fa9314e914dd74819c47e77#diff-7822a3eb7478100e4687627162508ec020c0de25ad5113d6d8c2994e5f9b6a0bR54)
> 
> I know it's temporary, but did you mean LNX? The prefix LIN is not listed: https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/system/Capabilities.html#version

You are correct; I'll fix that.

**Comment by rdrpenguin04** [comment]
> Could you rebase onto master? Thank you!

Rebasing was not working unfortunately; every single Ragnarok commit created many merge conflicts. If you know of a way to streamline the rebasing process, I could rewind and rebase. For now, I have a merge unfortunately.

File: core/src/avm2.rs (hunk 1 above)
Comment (rdrpenguin04) [reply core/src/avm2.rs]: Fixed in latest commit

**Comment by rdrpenguin04** [comment]
Adrian has done the rebase, this is now officially ready for review! :smile:

File: core/src/avm2/globals/string.rs [hunk 2]
```diff
@@ -118,6 +119,50 @@ fn char_code_at<'gc>(
...
 }
 
+/// Implements `String.split`
+fn split<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        let this = Value::from(this).coerce_to_string(activation)?;
```
[the remark this answers is not in the document]
Comment (rdrpenguin04) [reply core/src/avm2/globals/string.rs:129]: Added warnings

File: core/src/avm2/class.rs [hunk 3]
```diff
@@ -462,6 +462,31 @@ impl<'gc> Class<'gc> {
         }
     }
     #[inline(never)]
+    pub fn define_public_builtin_class_traits(
```
[the remark this answers is not in the document]
Comment (rdrpenguin04) [reply core/src/avm2/class.rs:465]: Renamed

**Review by adrian17 — APPROVED** [review]

**Pull request merged**